cellengine/resources/experiment.py

```diff
@@ -28,12 +34,108 @@
     format_split_gate,
     format_quadrant_gate,
 )
-from cellengine.utils.helpers import today_timestamp
+from cellengine.utils.helpers import (
+    CommentList,
+    datetime_to_timestamp,
+    timestamp_to_datetime,
+)
+
+
+@dataclass
+class Experiment(DataClassMixin):
+
+    name: str
+    annotation_validators: Dict[Any, Any]
+    annotation_name_order: List[str]
+    annotation_table_sort_columns: List[Union[str, int]]
+    created: datetime = field(
+        metadata=config(
+            encoder=datetime_to_timestamp,
+            decoder=timestamp_to_datetime,
+            mm_field=fields.DateTime(),
+        )
+    )
+    _comments: List[Dict[str, Any]] = field(metadata=config(field_name="comments"))
+    deep_updated: datetime = field(
+        metadata=config(
+            encoder=datetime_to_timestamp,
+            decoder=timestamp_to_datetime,
+            mm_field=fields.DateTime(),
+        )
+    )
+    per_file_compensations_enabled: bool
+    primary_researcher: Dict[str, Any]
+    revisions: List[Dict[str, Any]]
+    sorting_spec: Dict[Any, Any]
+    tags: List[str]
+    uploader: Dict[str, Any]
+    updated: datetime = field(
+        metadata=config(
+            encoder=datetime_to_timestamp,
+            decoder=timestamp_to_datetime,
+            mm_field=fields.DateTime(),
+        ),
+        default=ReadOnly(),
+    )  # type: ignore
+    _active_comp: Optional[Union[int, str]] = field(
+        default=None, metadata=config(field_name="activeCompensation")
+    )
+    deleted: Optional[datetime] = field(
+        default=None,
+        metadata=config(
+            encoder=lambda t: datetime_to_timestamp(t) if t else t,
+            decoder=lambda t: timestamp_to_datetime(t) if t else t,
+            mm_field=fields.DateTime(),
+        ),
+    )
+    data: Optional[Dict[Any, Any]] = field(
+        default=ReadOnly(optional=True)
+    )  # type: ignore
+    clone_source_experiment: Optional[str] = field(
+        default=ReadOnly(optional=True)
+    )  # type: ignore
+    locked: bool = field(default=ReadOnly())  # type: ignore
+    permissions: List[Dict[Any, Any]] = field(default=ReadOnly())  # type: ignore
+    retention_policy: Dict[str, Any] = field(default=ReadOnly())  # type: ignore
+    revision_source_experiment: Optional[str] = None
+    _id: str = field(
+        metadata=config(field_name="_id"), default=ReadOnly()
+    )  # type: ignore
+
+    def __repr__(self):
+        return f"Experiment(_id='{self._id}', name='{self.name}')"
+
+    @property
+    def comments(self):
+        """Get comments for experiment.
 
+        Defaults to overwrite; append new comments with
+        experiment.comments.append(dict) with the form:
+        dict = {"insert": "some text",
+        "attributes": {"bold": False, "italic": False, "underline": False}}.
```

Comment:
If this is a getter, these lines should probably be deleted.

Reply:
sorry, not pending, I just forgot to `-f` and didn't notice the push failure message

Reply:
I meant to ask, should the documentation about adding new comments be moved to the `@comments.setter` def. Do the documentation tools read the doc string off of the `@property` and ignore the `@comments.setter` though?